searcher.go

```diff
@@ -146,11 +146,15 @@ func (s *Searcher) SearchMonitoring(clusterID string) (Monitoring, error) {
 	return monitoring, nil
 }
 
-func (s *Searcher) searchError(tls *TLS, clusterID string, cert Cert, err error) error {
+func (s *Searcher) SearchTLS(clusterID string, cert Cert) (TLS, error) {
```

Comment:
The diff is weird. `searchError` got removed. I noticed it was not used for anything. Just dead code. I added `SearchTLS` though for abstraction purposes. That way we can easily fetch dedicated key pairs like the new `guestcluster` lib requires.